contacts/tests.py

```diff
@@ -1,3 +1,126 @@
+        seeker = create_seeker(user, new_city, start_date, min_rent, max_rent, roomates, rooms, about)
+        seeker.save()
+        return seeker
+
+    return _make_seeker
+
+
+@pytest.fixture
+def sample_apartment(new_city):
+    user3 = create_new_user("t3@m.com", "test3", "test", "1995-05-05", "testing")
+    apartment = create_apartment(user3, new_city, "Hatotahim 5", 2500, 2, 3, "2020-05-05")
+    apartment.save()
+    return apartment
+
+
+@pytest.fixture
+def sample_connection(make_seeker, sample_apartment):
+    user1 = create_new_user("t1@m.com", "test1", "test", "1995-05-05", "testing")
+    seeker = make_seeker(user1, "2020-05-05", 100, 1000, 2, 2, "Hello")
+    con = Connection(seeker=seeker, apartment=sample_apartment)
+    con.save()
+    return con
+
+
+@pytest.mark.django_db
+def test_add_connection(sample_connection):
+    assert sample_connection is not None
+    assert sample_connection.seeker.base_user.email == "t1@m.com"
+    assert sample_connection.apartment.owner.email == "t3@m.com"
+
+
+@pytest.mark.django_db
+def test_find_connection(sample_connection):
+    temp_connection = Connection.objects.get(seeker=sample_connection.seeker, apartment=sample_connection.apartment)
+    assert temp_connection == sample_connection
+
+
+@pytest.mark.django_db
+def test_invalid_users_connection(make_seeker):
+    user1 = create_new_user("t1@m.com", "test1", "test", "1995-05-05", "testing")
+    seeker1 = make_seeker(user1, "2020-05-05", 100, 1000, 2, 2, "Hello")
+    user2 = create_new_user("t2@m.com", "test2", "test", "1995-05-05", "testing")
+    seeker2 = make_seeker(user2, "2020-05-05", 300, 3000, 3, 3, "Hi")
+    with pytest.raises(ValueError):
+        with transaction.atomic():
+            Connection(seeker=seeker1, apartment=seeker2).save()
+
+
+@pytest.mark.django_db
+def test_invalid_duplicate_connection(sample_connection):
+    with pytest.raises(IntegrityError):
+        with transaction.atomic():
+            Connection(seeker=sample_connection.seeker, apartment=sample_connection.apartment).save()
+
+
+@pytest.mark.django_db
+def test_multipile_connection(sample_connection, make_seeker):
+    user2 = create_new_user("t2@m.com", "test2", "test", "1995-05-05", "testing")
+    seeker2 = make_seeker(user2, "2020-05-05", 300, 3000, 3, 3, "Hi")
+    Connection(seeker=seeker2, apartment=sample_connection.apartment).save()
+    amount = Connection.objects.filter(apartment=sample_connection.apartment).count()
+    expected = 2
+    assert amount == expected
+
+
+@pytest.mark.django_db
+def create_new_user(email, f_name, l_name, b_date, passw):
+    user1 = User(
+        email=email,
+        first_name=f_name,
+        last_name=l_name,
+        birth_date=b_date,
+        password=passw
+    )
+    user1.save()
+    return user1
+
+
+def create_city():
+    city = City(cityName='Tel Aviv')
+    return city
+
+
+def create_seeker(user, city, s_date, min_r, max_r, roomates, rooms, about):
```

Comment:
This can be included in the fixture - no need for a separate function.

Reply:
Included in a fixture